Nos/Views/AppView.swift

```diff
@@ -147,8 +148,34 @@ struct AppView: View {
             UITabBar.appearance().unselectedItemTintColor = .secondaryTxt
             UITabBar.appearance().tintColor = .primaryTxt
         }
+        .sheet(isPresented: $showNIP05Wizard) {
+            CreateUsernameWizard(isPresented: $showNIP05Wizard)
+        }
+        .task(presentNIP05SheetIfNeeded)
         .accentColor(.primaryTxt)
     }
+
+    @Sendable private func presentNIP05SheetIfNeeded() async {
+        // Sleep for half a second
+        try? await Task.sleep(nanoseconds: 500_000_000)
+        guard let author = currentUser.author, let npub = author.npubString else {
+            return
+        }
+        // We store the npub for the user we last presented the sheet for so that
+        // the behavior resets if the user creates a new account
+        let key = "AppView_presentedNIP05SheetFor"
+        let didPresentSheetInPast = UserDefaults.standard.string(forKey: key)
+        var shouldShowSheet: Bool
+        if let didPresentSheetInPast {
+            shouldShowSheet = didPresentSheetInPast != npub && author.nip05 == nil
+        } else {
+            shouldShowSheet = author.nip05 == nil
+        }
+        if shouldShowSheet {
+            showNIP05Wizard = true
+            UserDefaults.standard.setValue(npub, forKey: key)
```

Comment:
Do we generally just use `UserDefaults.standard` directly like this? It's sort of a hidden dependency; at least, hidden outside of this func. Do we have a way to specify it as a `@Dependency`? I still don't fully know how all of that works.